### PR TITLE
update pairselection to include delta R check

### LINKED ISSUE #118: [BUG] DeltaR requirement in Pair Selection
**Describe the bug**
The DeltaR Requriement for the separation of the two pair candidates is not included in the algorithm

**Expected behavior**
It should be included

### PR DESCRIPTION
**Pull request by harrypuuter**
Resolves #118

### CODE REVIEW

File: src/pairselection.hxx [hunk 1]
```diff
@@ -387,30 +395,62 @@ auto PairSelectionAlgo() {
...
+                    ->debug("MuonPt = {} , TauPt = {} ", muon.Pt(), tau.Pt());
+                Logger::get("PairSelection")
+                    ->debug("MuonPt = {} , TauPt = {} ", muon_pt[muonindex],
+                            tau_pt[tauindex]);
+                Logger::get("PairSelection")
+                    ->debug("MuonIso = {} , TauIso = {} ", muon_iso[muonindex],
+                            tau_iso[tauindex]);
+                selected_pair = {static_cast<int>(muonindex),
+                                 static_cast<int>(tauindex)};
+                break;
```
Comment (nshadskiy) [inline src/pairselection.hxx:449]: What happens if all `sorted_pairs` fail the deltaR criteria. Does `selected_pair` have default values?

File: src/pairselection.hxx [hunk 2]
```diff
@@ -422,15 +462,30 @@ auto PairSelectionAlgo() {
...
+    - tau_iso
+    - muon_pt
+    - muon_eta
+    - muon_phi
+    - muon_mass
+    - muon_iso
+    - tau_mask containing the flags whether the tau is a good tau or not
+    - muon_mask containing the flags whether the muon is a good tau or not
  * @param pairname name of the new column containing the pair index
+ * @param mindeltaR the seperation between the two muons has to be larger than
```
Comment (nshadskiy) [inline src/pairselection.hxx:479]: "between muon and tau"

File: src/pairselection.hxx (hunk 1 above)
Comment (harrypuuter) [reply src/pairselection.hxx:449]: you are correct, this was missing !

File: src/pairselection.hxx (hunk 2 above)
Comment (harrypuuter) [reply src/pairselection.hxx:479]: fixed

File: src/pairselection.hxx (hunk 1 above)
Comment (harrypuuter) [reply src/pairselection.hxx:449]: Should be fixed now

File: src/pairselection.hxx [hunk 3]
```diff
@@ -559,41 +659,59 @@ auto ZBosonPairSelectionAlgo() {
...
- * @param input_vector vector of strings containing the columns needed for the
- * alogrithm
+ * @param input_vector vector of strings containing the columns
+ * needed for the alogrithm. For the muon pair selection the required paramters
+ are:
+    - muon_pt
+    - muon_eta
+    - muon_phi
+    - muon_mass
+    - muon_mask containing the flags whether the muon is a good tau or not
```
Comment (nshadskiy) [inline src/pairselection.hxx:678]: "muon is a good muon"
The same in the `mutau` namespace (line 477)

**Review by nshadskiy — APPROVED** [review]
Looks good!

**Pull request merged**